### DISCUSSION TITLE
How to add “Automatic export"  location after it had been deleted?

### DISCUSSION THREAD

**Discussion by earthmoon** [retorquere/zotero-better-bibtex#2514] [Q&A]
Hello,

I had once deleted the "Automatic export" setting, so it is now:
![image](https://github.com/retorquere/zotero-better-bibtex/assets/45897471/d6282d77-6115-4a33-8b87-5a6d3dccbf5a)

If I need to restore automatic export, what should I do? I can't restore it now.

Thanks.

**Comment by retorquere** [comment]
Make a new export, and check "keep updated", the same way you set it up before.

**Reply by earthmoon** [reply]
> Make a new export, and check "keep updated", the same way you set it up before.

Why don't let it can be set in the "Automatic export" in the Better BitTeX reference there? Getting the information from making a new export is a way, setting it manually there can be another way. It is not convenient only by making a new export.

**Reply by retorquere** [reply]
Because I'd have to re-implement the Zotero collection view to allow you to pick the library/group/collection to export, and I see no value to doing that work when the only thing it would save you is one mouse click.

**Comment by earthmoon** [comment]
>Because I'd have to re-implement the Zotero collection view to allow you to pick the library/group/collection to export

May you explain this sentence. I can't understand it. Sorry.

**Reply by retorquere** [reply]
Can you instead describe how it would work when you can set up an export from the preferences? Including what you would see on screen, step by step. You want to export a collection or a library, because that's what auto-export is built around. What would be the first step? And the next?

**Reply by earthmoon** [reply]
> Can you instead describe how it would work when you can set up an export from the preferences? Including what you would see on screen, step by step. You want to export a collection or a library, because that's what auto-export is built around. What would be the first step? And the next?

Maybe I have guessed what you had expressed. Let me say it slowly.

1. I exported a library to the location of `C:\Downloads\one.bib`;
2. Then I exported an collection to the location of `C:\Downloads\two.bib`;
3. Then in the **Automatic export** of the Better BibTeX setting, if I choose `C:\Downloads\one.bib`,  it will automatically export the library;
4. Then  in the **Automatic export** of the Better BibTeX setting, if I choose `C:\Downloads\two.bib`,  it will automatically export the collection;

Am I right?

**Reply by retorquere** [reply]
No, in step 1. and 2. you just check the checkbox "Keep updated". There's nothing else to do. You only have to go into the preferences if you want to remove an auto-export or change its settings.

**Reply by earthmoon** [reply]
> o, in step 1. and 2. you just check the checkbox "Keep updated".

I have tested and found these:

1. If don't check the checkbox "Keep updated" and export it, I can't find it in the  **Automatic export** of the Better BibTeX setting, so there is need to check the checkbox "Keep updated";
2. If there are three locations in the   **Automatic export** of the Better BibTeX setting, can it export three locations simultaneously?

**Reply by retorquere** [reply]
> If don't check the checkbox "Keep updated" and export it, I can't find it in the Automatic export of the Better BibTeX setting, so there is need to check the checkbox "Keep updated";

Yes, that is how it works.

> After checking the checkbox "Keep updated" and exporting it from one library and one collection seperateley, I found that even in the exporting of one collection, the exported file contains all reference not just the ones in the collection;

That would be a bug then, and you can open an issue for that. I cannot reproduce this.

> If there are three locations in the Automatic export of the Better BibTeX setting, can it export three locations simultaneously?

They will all be triggered, but technically, they will run in series, not in parallel.

**Reply by earthmoon** [reply]
> > If don't check the checkbox "Keep updated" and export it, I can't find it in the Automatic export of the Better BibTeX setting, so there is need to check the checkbox "Keep updated";
> 
> Yes, that is how it works.

After having done these steps, I need to uncheck the "Keep updated" if I want to use **Automatic export** of the Better BibTeX setting, or they do the same thing. Am I right?

> > After checking the checkbox "Keep updated" and exporting it from one library and one collection seperateley, I found that even in the exporting of one collection, the exported file contains all reference not just the ones in the collection;
> 
> That would be a bug then, and you can open an issue for that. I cannot reproduce this.
> 
> > If there are three locations in the Automatic export of the Better BibTeX setting, can it export three locations simultaneously?
> 
> They will all be triggered, but technically, they will run in series, not in parallel.

I have tested that  if the three locations have different setting in the **Automatic export** of the Better BibTeX setting, such as one with "idle" one with "on change" and one with "paused", they wouldn't work in series.

**Reply by retorquere** [reply]
> After having done these steps, I need to uncheck the "Keep updated" if I want to use **Automatic export** of the Better BibTeX setting, or they do the same thing. Am I right?

I'm not sure we're using the terminology in the same way. Automatic export is the way exports are kept updated. They mean essentially the same thing. So if you uncheck "keep updated", you don't get a corresponding automatic export.

> I have tested that if the three locations have different setting in the **Automatic export** of the Better BibTeX setting, such as one with "idle" one with "on change" and one with "paused", they wouldn't work in series.

If you pause auto exports, no auto-exports will happen until you change it back. That's what it means to pause the auto-exports. Idle means that they will be paused until Zotero has sat still for a while, so while you're actively using Zotero, no auto-exports will happen. On change means as soon as possible, so then you should see the auto-exports ran.

If that's not happening for you, you need to open an issue.

**Reply by earthmoon** [reply]
> I'm not sure we're using the terminology in the same way. Automatic export is the way exports are kept updated. They mean essentially the same thing. So if you uncheck "keep updated", you don't get a corresponding automatic export.

But if the "Keep update" is unchecked, the Automatic export in the Better BibTeX still works. That is what I found. May you try it?

>If you pause auto exports, no auto-exports will happen until you change it back. That's what it means to pause the auto-exports. Idle means that they will be paused until Zotero has sat still for a while, so while you're actively using Zotero, no auto-exports will happen. On change means as soon as possible, so then you should see the auto-exports ran.

So what you said  "They will all be triggered, but technically, they will run in series, not in parallel." only means that the three location can do the same action with the Automatic export in the Better BibTeX setting. Am I right?

**Comment by retorquere** [comment]
> But if the "Keep update" is unchecked, the Automatic export in the Better BibTeX still works. That is what I found. May you try it?

You export with "keep updated" to create a **new** auto-export. After creation, it keeps exporting when you make a relevant change. If you export again after that without "keep updated", that just has no effect on any auto-export. If you export again with "keep updated", you replace an existing export (if you choose the same file), or you create 2nd, 3rd, etc one.

> means that the three location can do the same action with the Automatic export in the Better BibTeX setting. Am I right?

Yes.

**Reply by earthmoon** [reply]
I am very confused what you said. 

1. "keep update" is a function from Better BibTeX, as I can't see this choice if I don't choose any export format that begins with "Better";
2. If the "keep update" is checked, I can change the update state in the Automatic export in the Better BibTeX setting, am I right?
3. If the "keep update" is checked, I choose "paused" in the Automatic export in the Better BibTeX setting, and then the library or collection wouldn't be exported automatically. Am I right?

If 1,2 and 3 are right, I think it is better to  change the name of  "keep update" to the name "automatic export in Better BibTeX". As if hadn't talked with you, I thought "keep update" is a choice from Zotero not from Better BibTeX and if select "paused" in the Automatic export in the Better BibTeX setting the references wouldn't be updated.

**Reply by earthmoon** [reply]
>>means that the three location can do the same action with the Automatic export in the Better BibTeX setting. Am I right?

>Yes.

I feel this is dangerous. For example, I exported the first collection to location A with "keep update" checked",  exported the second collection to location B with "keep update" and exported the third collection to location C with "keep update“. Then I didn't need to update any of three locations, so I selected the  "paused" in the Better BibTeX setting and all three location wouldn't be updated now. Then I wanted to the location B be updated, so if I didn't delete location A and location C, choose "On Change", and any changes in the first and the third collections would also cause location A and location C updated. If what I said is right, I think most users don't know this.

**Reply by retorquere** [reply]
>     1. "keep update" is a function from Better BibTeX, as I can't see this choice if I don't choose any export format that begins with "Better";

Correct.
 
>     2. If the "keep update" is checked, I can change the update state in the Automatic export in the Better BibTeX setting, am I right?

No. "Keep updated" only creates new auto-exports. 
 
>     3. If the "keep update" is checked, I choose "paused" in the Automatic export in the Better BibTeX setting, and then the library or collection wouldn't be exported automatically. Am I right?

Until you change it back, yes.

> If 1,2 and 3 are right, I think it is better to change the name of "keep update" to the name "automatic export in Better BibTeX". As if hadn't talked with you, I thought "keep update" is a choice from Zotero not from Better BibTeX and if select "paused" in the Automatic export in the Better BibTeX setting the references wouldn't be updated.

I'll consider this. But you're the first to report this problem, and automated exports have existed for years.

> I feel this is dangerous.

That's a pretty bold claim. Dangerous how?

> For example, I exported the first collection to location A with "keep update" checked", exported the second collection to location B with "keep update" and exported the third collection to location C with "keep update“.

No problem.

> Then I didn't need to update any of three locations, so I selected the "paused" in the Better BibTeX setting and all three location wouldn't be updated now.

If you didn't need to update any of these three, why did you set up "keep updated" for them?

> Then I wanted to the location B be updated, so if I didn't delete location A and location C, choose "On Change", and any changes in the first and the third collections would also cause location A and location C updated.

So keep them paused and hit "export now" on B. 

I understand now what you want (I think), but I don't know why you would want this, and this is simply not how auto-export works. Exports update when any of their own items change. If an item changes that is only in A and C, B is not exported at that time. If you unpause updates, pending updates will be written, yes. If you don't want them to update, don't add them as automatic updates. Pause does not mean ignore, the *exports* are paused, not the change accumulation. 

> If what I said is right, I think most users don't know this.

Because you know most of my users?

**Reply by earthmoon** [reply]
Maybe my feeling is wrong. Good job!
